src/main/java/uk/ac/ebi/eva/contigalias/repo/AssemblyRepository.java

```diff
@@ -31,21 +31,21 @@
 public interface AssemblyRepository extends JpaRepository<AssemblyEntity, Long>,
         JpaSpecificationExecutor<AssemblyEntity> {
 
-    default Optional<AssemblyEntity> findAssemblyEntityByAccession(String accession) {
-        return this.findAssemblyEntityByGenbankOrRefseq(accession, accession);
+    default Page<AssemblyEntity> findAssemblyEntityByAccession(String accession, Pageable pageable) {
+        return this.findAssemblyEntityByGenbankOrRefseq(accession, accession, pageable);
     }
 
-    Optional<AssemblyEntity> findAssemblyEntityByGenbankOrRefseq(String genbank, String refseq);
+    Page<AssemblyEntity> findAssemblyEntityByGenbankOrRefseq(String genbank, String refseq, Pageable pageable);
 
     long count();
 
     Optional<AssemblyEntity> findTopByIdNotNullOrderById();
 
-    Optional<AssemblyEntity> findAssemblyEntityByGenbank(String genbank);
+    Page<AssemblyEntity> findAssemblyEntityByGenbank(String genbank, Pageable pageable);
 
-    Optional<AssemblyEntity> findAssemblyEntityByRefseq(String refseq);
+    Page<AssemblyEntity> findAssemblyEntityByRefseq(String refseq, Pageable pageable);
```

Comment:
In my other comment about the paging strategy I referred to this. I think this should return an Optional and not ask for a pageable.

Reply:
Created a new issue for this #52